<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>707</width>
    <height>473</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>SCRT Control Center</string>
  </property>
  <property name="documentMode">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="sizeConstraint">
     <enum>QLayout::SetDefaultConstraint</enum>
    </property>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>Toolbar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionConnect"/>
   <addaction name="separator"/>
   <addaction name="controlSelfTest"/>
   <addaction name="openHelp"/>
  </widget>
  <action name="actionConnect">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../Assets/Resources.qrc">
     <normaloff>:/Icons/22x22/Connect.png</normaloff>
     <normalon>:/Icons/22x22/Connect.png</normalon>:/Icons/22x22/Connect.png</iconset>
   </property>
   <property name="text">
    <string>Connect</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="controlSelfTest">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../Assets/Resources.qrc">
     <normaloff>:/Icons/22x22/Bit.png</normaloff>:/Icons/22x22/Bit.png</iconset>
   </property>
   <property name="text">
    <string>Run Built-in Self-Test</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+T</string>
   </property>
  </action>
  <action name="openHelp">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../Assets/Resources.qrc">
     <normaloff>:/Icons/22x22/Help.png</normaloff>:/Icons/22x22/Help.png</iconset>
   </property>
   <property name="text">
    <string>Help</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+H</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="../Assets/Resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionConnect</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>openConnection()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>101</x>
     <y>313</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openHelp</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>openHelpDialog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>263</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>controlSelfTest</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>onControlSelfTest(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>353</x>
     <y>236</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>savePositions()</slot>
  <slot>calibrateFeedbacks()</slot>
  <slot>openSerialPort()</slot>
  <slot>openSettings()</slot>
  <slot>updateServoPosition(unsigned,unsigned)</slot>
  <slot>calibrateServo()</slot>
  <slot>openHelpDialog()</slot>
  <slot>onControlSelfTest(bool)</slot>
  <slot>openSocket()</slot>
  <slot>openConnection()</slot>
 </slots>
</ui>
